irohad/network/impl/async_grpc_client.hpp

```diff
@@ -32,7 +32,9 @@ namespace iroha {
     template <typename Response>
     class AsyncGrpcClient {
      public:
-      AsyncGrpcClient() : thread_(&AsyncGrpcClient::asyncCompleteRpc, this) {}
+      explicit AsyncGrpcClient(logger::Logger &&log)
+          : thread_(&AsyncGrpcClient::asyncCompleteRpc, this),
+            log_(std::move(log)) {}
```

Comment:
Same as before - it's not clear why not to create logger just here.

Reply:
Related classes are children of `AsyncGrpcClient`, so they share exactly same logger. But now `AsyncGrpcClient` issues are logged related to the child, e.g issue at `asyncCompleteRpc` may have prefix of child (for example `NetworkImpl`) or something more meaningful, than just `AsyncGrpcClient`